include/onnxruntime/core/session/onnxruntime_cxx_inline.h

```diff
@@ -929,6 +929,22 @@ inline void SessionImpl<T>::Run(const RunOptions& run_options, const IoBinding&
   ThrowOnError(GetApi().RunWithBinding(this->p_, run_options, io_binding));
 }
 
+inline void CallbackBridge(void* user_data, OrtValue** outputs, size_t num_outputs, OrtStatusPtr status) {
+  RunAsyncCallbackStdFn* callback = reinterpret_cast<RunAsyncCallbackStdFn*>(user_data);
+  std::vector<Ort::Value> output_values;
+  for (size_t ith = 0; ith < num_outputs; ++ith) {
+    output_values.emplace_back(outputs[ith]);
+  }
+  (*callback)(output_values, Ort::Status{status});
```

Comment:
Won't output_values go out of scope?

Reply:
Apparently not - why would it be?

Reply:
This should be fine. the callback should be (or, is expected to be) a synchronized function